rust/operator-binary/src/superset_controller.rs

```diff
@@ -91,6 +107,18 @@ pub enum Error {
     ObjectMissingMetadataForOwnerRef {
         source: stackable_operator::error::Error,
     },
+    #[snafu(display("superset only supports a single authentication method"))]
+    MultipleAuthenticationMethods,
+    #[snafu(display("failed to retrieve authentication class {}", authentication_class))]
+    AuthenticationClassRetrieval {
+        source: stackable_operator::error::Error,
+        authentication_class: String,
+    },
```

Comment:
```suggestion
    #[snafu(display("failed to retrieve {}", authentication_class))]
    AuthenticationClassRetrieval {
        source: stackable_operator::error::Error,
        authentication_class: ObjectRef<AuthenticationClass>,
    },
```

Reply:
The problem that i see here is that IMHO we don't have an `AuthenticationClass` instance here to pass to the Error.
The only thing that we have is the name of the AuthenticationClass and in the case it does not exist the only information we have is the name of the AuthenticationClass that the user provided

Reply:
The name (and NS, if applicable) is all we need to build an `ObjectRef` with the `ObjectRef::<Kind>::new()` constructor.

Reply:
Ah thanks! I though the ObjectRef<> is a Rust thing but its a kubernetes reference